### PR TITLE
add e2e test for scaling workers

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/giantswarm/issues/2023

Adds e2e test for scaling guest cluster up and down. The changes in the custom object are done through patching with JSON-patch.

### CODE REVIEW

File: integration/framework.go [hunk 1]
```diff
@@ -12,16 +13,26 @@ import (
...
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	giantclientset "github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/e2e-harness/pkg/harness"
+	"github.com/giantswarm/microerror"
```
Comment (rossf7) [inline integration/framework.go]: <3

File: integration/framework.go [hunk 2]
```diff
@@ -346,14 +357,14 @@ func (f *framework) WaitForGuestReady() error {
 		return microerror.Mask(err)
 	}
 
-	if err := f.waitForNodesUp(); err != nil {
+	if err := f.WaitForNodesUp(minimunNodesReady); err != nil {
```
Comment (rossf7) [inline integration/framework.go]: Nitpick: minimum

File: integration/operator_test.go [hunk 3]
```diff
@@ -244,3 +279,45 @@ func writeAWSResourceValues() error {
...
 	return nil
 }
+
+func numberOfWorkers(clusterName string) (int, error) {
+	cluster, err := f.AWSCluster(clusterName)
+	if err != nil {
+		return 0, microerror.Mask(err)
+	}
+
+	return len(cluster.Spec.AWS.Workers), nil
```
Comment (rossf7) [inline integration/operator_test.go]: Could you use `keyv2.WorkerCount` here?

File: integration/framework.go (hunk 2 above)
Comment (fgimenez) [reply integration/framework.go]: Done

File: integration/operator_test.go (hunk 3 above)
Comment (fgimenez) [reply integration/operator_test.go]: Sure, I've added `keyv2.MasterCount` too.

**Pull request merged**